### ISSUE TITLE
IDurableOrchestrationClient throws TimeoutException with message "Client request timed out."

### ISSUE THREAD

**Issue by sqbing** [microsoft/durabletask-netherite#331]
We're doing some Netherite performance tests on Azure. The function execution chain looks like,
```
Timer trigger function -> Event Grid -> Event Hub ->Event Hub trigger function -> Orchestrator Function -> Activity Function
```
- The `Timer trigger function` constantly generates 400 events per second.
- The `Event Hub trigger function` creates an orchestrator function instance per event.

The whole test runs smooth for around 7 hours. The `StartNewAsync` completes within 450 ms on average.
However, around 2023-10-31T16:53:56.5537109Z, the `StartNewAsync` latency increased significantly.

Please let me know if you need more information.

## Code of `Event Hub trigger function`
```
        [FunctionName(FunctionName)]
        [Disable($"Disable{FunctionName}")]
        public async Task Run(
            [EventHubTrigger("DurableNotification", Connection = "DurableNotificationConnectionString")] EventData[] events,
            ILogger log,
            [DurableClient] IDurableOrchestrationClient orchestrationClient)
        {
            var exceptions = new List<Exception>();

            IEnumerable<Task> tasks = events.Select(x => StartOrchFunction(orchestrationClient));
            
            Stopwatch stopwatch = Stopwatch.StartNew();
            try
            {
                await Task.WhenAll(tasks);

                telemetryClient.GetMetric("EventHubTriggerFunction Success").TrackValue(1);
            }
            catch (Exception ex)
            {
                log.LogError(ex, "exception when group starting orchestrator function");

                telemetryClient.GetMetric("EventHubTriggerFunction Failure").TrackValue(1);
            }
            finally
            {
                stopwatch.Stop();

                telemetryClient.GetMetric("EventHubTriggerFunction Duration").TrackValue(stopwatch.Elapsed.TotalMilliseconds);
            }
        }

        private async Task StartOrchFunction(IDurableOrchestrationClient client)
        {
            Stopwatch stopwatch = new();
            stopwatch.Start();
            await client.StartNewAsync(NewDurableTestFunction.OrchFunctionName, null);
            stopwatch.Stop();
            telemetryClient.GetMetric("StartNewAsync").TrackValue(stopwatch.Elapsed.TotalMilliseconds);
        }
```

## Stack trace (partial):
filename: /_/src/DurableTask.Netherite/OrchestrationService/Client.cs line: 469
filename: /_/src/DurableTask.Netherite/OrchestrationService/NetheriteOrchestrationService.cs line: 594
filename: /_/src/DurableTask.Core/TaskHubClient.cs line: 614
filename: D:\a\_work\1\s\src\WebJobs.Extensions.DurableTask\ContextImplementations\DurableClient.cs line: 614
filename: EventHubTriggerFunction.cs line: 62

## Timestamp: around _2023-11-01T01:55:52.0098317Z_

**Comment by sebastianburckhardt** [comment]
Thanks for providing the details. I took a look at the internal telemetry. It looks like the long delays are caused by a partition outage which is caused by a bug in 1.4.0 (the error messages claim that there are out-of-order EH packages). We have since then fixed this bug (https://github.com/microsoft/durabletask-netherite/pull/302) so the recommendation is to try again with the latest Netherite version (1.4.1).
